tests/test_mathematical.py

```diff
@@ -392,14 +392,37 @@ def test_nancumsum(array):
     [[[1.0, -1.0], [0.1, -0.1]], [-2, -1, 0, 1, 2]],
     ids=["[[1., -1.], [0.1, -0.1]]", "[-2, -1, 0, 1, 2]"],
 )
-@pytest.mark.parametrize("dtype", get_all_dtypes(no_bool=True, no_complex=True))
+@pytest.mark.parametrize("dtype", get_all_dtypes())
 def test_negative(data, dtype):
     np_a = numpy.array(data, dtype=dtype)
     dpnp_a = dpnp.array(data, dtype=dtype)
 
-    result = dpnp.negative(dpnp_a)
-    expected = numpy.negative(np_a)
-    assert_allclose(result, expected)
+    if dtype == dpnp.bool:
+        with pytest.raises(TypeError):
+            dpnp.negative(dpnp_a)
```

Comment:
it would be better to separate that to a test without parametrized data, which will check only the exception with boolean type of some input array